src/storage/src/table/state_table.rs

```diff
@@ -113,171 +116,109 @@ impl<S: StateStore> StateTable<S> {
         Ok(())
     }
 
-    pub async fn iter(&self, epoch: u64) -> StorageResult<StateTableRowIter<'_, S>> {
-        let mem_table_iter = self.mem_table.buffer.iter().peekable();
-        StateTableRowIter::new(
+    pub async fn iter(
+        &self,
+        epoch: u64,
+    ) -> StorageResult<Pin<Box<impl Stream<Item = StorageResult<Option<Row>>> + Send + '_>>> {
```

Comment:
We may directly use opaque type here?
```suggestion
    ) -> StorageResult<impl Stream<Item = StorageResult<Option<Row>>> + '_> {
```

Reply:
Yes, we could, the caller `Box::pin` it.